superset/common/query_context_processor.py

```diff
@@ -600,7 +600,15 @@ def get_payload(
             set_and_log_cache(
                 cache_manager.cache,
                 cache_key,
-                {"data": self._query_context.cache_values},
+                {
+                    "data": {
+                        # setting form_data into query context cache value as well
+                        # so that it can be used to reconstruct form_data field
+                        # for query context object when reading from cache
+                        "form_data": self._query_context.form_data,
+                        **self._query_context.cache_values,
```

Comment:
to be used by this later https://github.com/apache/superset/blob/master/superset/security/manager.py#L1898C1-L1898C1

Reply:
@villebro do you have any thoughts on this?

Reply:
@zephyring from where does the raise_for_access get called when it is passing in the query context from cache?

Reply:
The first call to `/chart/data` endpoint will set the query context into cache and return async job metadata.
Then later once job is done frontend will receive a job ready event with the result_url something like (`/api/v1/chart/data/qc-2772209e0a4b1ca513375118855b27af`).
Then it uses that to call `chart/data/<cache>` endpoint which invokes the [ChartCommand](https://github.com/apache/superset/blob/master/superset/charts/data/api.py#L307) and the command validation will call the raise_for_access